autocompletion on invalid earthfile

Consider an Earthfile like:
```
FROM python:3

test:
  RUN111 echo test step

build:
  RUN echo build step
```

Then if you run `earth +<tab><tab>`, you get a panic:
```
	panic: interface conversion: interface is nil, not antlr.ParserRuleContext

goroutine 1 [running]:
github.com/earthly/earthly/earthfile2llb/parser.(*EarthParser).EarthFile.func2(0xc0001c98a0, 0xc0003f4038)
	/earthly/earthfile2llb/parser/earth_parser.go:452 +0x13f
panic(0xda7380, 0xc000467260)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/antlr/antlr4/runtime/Go/antlr.(*BailErrorStrategy).Recover(0xc0003f4020, 0x103be60, 0xc00040c870, 0x101c640, 0xc0003f4550)
	/go/pkg/mod/github.com/antlr/antlr4@v0.0.0-20200225173536-225249fdaef5/runtime/Go/antlr/error_strategy.go:741 +0x90
github.com/antlr/antlr4/runtime/Go/antlr.(*BailErrorStrategy).RecoverInline(0xc0003f4020, 0x103be60, 0xc00040c870, 0xc0003f44f0, 0x21)
	/go/pkg/mod/github.com/antlr/antlr4@v0.0.0-20200225173536-225249fdaef5/runtime/Go/antlr/error_strategy.go:750 +0x6d
github.com/antlr/antlr4/runtime/Go/antlr.(*BaseParser).Match(0xc00040c870, 0xffffffffffffffff, 0x21, 0x2)
	/go/pkg/mod/github.com/antlr/antlr4@v0.0.0-20200225173536-225249fdaef5/runtime/Go/antlr/parser.go:146 +0xe8
github.com/earthly/earthly/earthfile2llb/parser.(*EarthParser).EarthFile(0xc0003f4038, 0x103f6a0, 0xc000402320)
	/earthly/earthfile2llb/parser/earth_parser.go:533 +0x6b4
github.com/earthly/earthly/earthfile2llb.newEarthfileTree(0xed3d00, 0x9, 0x10232c0, 0xc0003f4018, 0x102c100, 0xc0003f4020, 0xc0001c99b8, 0x58159f, 0xed3d00, 0x9)
	/earthly/earthfile2llb/earthfile2llb.go:173 +0x213
github.com/earthly/earthly/earthfile2llb.GetTargets(0xed3d00, 0x9, 0x2, 0xed3d00, 0x9, 0x1, 0xc000402240)
	/earthly/earthfile2llb/earthfile2llb.go:178 +0xd4
github.com/earthly/earthly/autocomplete.getPotentialTarget(0xc00003c2d0, 0x1, 0xece692, 0x1, 0x0, 0x7fcf6cbf7d98, 0x0)
	/earthly/autocomplete/complete.go:81 +0x1ae
github.com/earthly/earthly/autocomplete.GetPotentials(0xc00003c2ca, 0x7, 0x7, 0xc000422000, 0x15, 0x20, 0xc0004040c0, 0x3, 0x4, 0xc0003fa000, ...)
	/earthly/autocomplete/complete.go:177 +0x70c
main.(*earthApp).autoComplete(0xc0003fa000)
	/earthly/cmd/earth/main.go:122 +0x2ac
main.main()
	/earthly/cmd/earth/main.go:252 +0x35d
panic: interface conversion: interface is nil, not antlr.ParserRuleContext [recovered]
	panic: interface conversion: interface is nil, not antlr.ParserRuleContext

```

--- a/cmd/earth/main.go
+++ b/cmd/earth/main.go
@@ -98,25 +98,33 @@ func (app *earthApp) autoComplete() {
 		return
 	}
 
+	err := app.autoCompleteImp()
+	if err != nil {
+		errToLog := err
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			os.Exit(1)
+		}
+		logDir := filepath.Join(homeDir, ".earthly")
+		logFile := filepath.Join(logDir, "autocomplete.log")
+		err = os.MkdirAll(logDir, 0755)
+		if err != nil {
+			os.Exit(1)
+		}
+		f, err := os.OpenFile(logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0755)
+		if err != nil {
+			os.Exit(1)
+		}
+		fmt.Fprintf(f, "error during autocomplete: %s\n", errToLog)
+		os.Exit(1)
+	}
+	os.Exit(0)
+}
+
+func (app *earthApp) autoCompleteImp() (err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			homeDir, err := os.UserHomeDir()
-			if err != nil {
-				os.Exit(1)
-			}
-			logDir := filepath.Join(homeDir, ".earthly")
-			logFile := filepath.Join(logDir, "autocomplete.log")
-			err = os.MkdirAll(logDir, 0755)
-			if err != nil {
-				os.Exit(1)
-			}
-			f, err := os.OpenFile(logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0755)
-			if err != nil {
-				os.Exit(1)
-			}
-			fmt.Fprintf(f, "error during autocomplete: %v\nstacktrace: %v\n\n", r, string(debug.Stack()))
-
-			os.Exit(1)
+			err = fmt.Errorf("recovered panic in autocomplete %s: %s", r, debug.Stack())
 		}
 	}()
 
@@ -125,7 +133,7 @@ func (app *earthApp) autoComplete() {
 
 	compPointInt, err := strconv.ParseUint(compPoint, 10, 64)
 	if err != nil {
-		panic(err)
+		return err
 	}
 
 	flags := []string{}
@@ -144,13 +152,13 @@ func (app *earthApp) autoComplete() {
 
 	potentials, err := autocomplete.GetPotentials(compLine, int(compPointInt), flags, commands)
 	if err != nil {
-		panic(err)
+		return err
 	}
 	for _, p := range potentials {
 		fmt.Printf("%s\n", p)
 	}
 
-	os.Exit(0)
+	return err
 }
 
 func (app *earthApp) insertBashCompleteEntry() error {

--- a/cmd/earth/main.go
+++ b/cmd/earth/main.go
@@ -13,6 +13,7 @@ import (
 	"path/filepath"
 	"regexp"
 	"runtime"
+	"runtime/debug"
 	"strconv"
 	"strings"
 	"syscall"
@@ -97,6 +98,28 @@ func (app *earthApp) autoComplete() {
 		return
 	}
 
+	defer func() {
+		if r := recover(); r != nil {
+			homeDir, err := os.UserHomeDir()
+			if err != nil {
+				os.Exit(1)
+			}
+			logDir := filepath.Join(homeDir, ".earthly")
+			logFile := filepath.Join(logDir, "autocomplete.log")
+			err = os.MkdirAll(logDir, 0755)
+			if err != nil {
+				os.Exit(1)
+			}
+			f, err := os.OpenFile(logFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0755)
+			if err != nil {
+				os.Exit(1)
+			}
+			fmt.Fprintf(f, "error during autocomplete: %v\nstacktrace: %v\n\n", r, string(debug.Stack()))
+
+			os.Exit(1)
+		}
+	}()
+
 	compLine := os.Getenv("COMP_LINE")   // full command line
 	compPoint := os.Getenv("COMP_POINT") // where the cursor is
 
@@ -121,8 +144,7 @@ func (app *earthApp) autoComplete() {
 
 	potentials, err := autocomplete.GetPotentials(compLine, int(compPointInt), flags, commands)
 	if err != nil {
-		//panic(err) // can't display error or it will show up under tab-completion
-		os.Exit(1)
+		panic(err)
 	}
 	for _, p := range potentials {
 		fmt.Printf("%s\n", p)

